### PR TITLE
Remove CircleCI

### PR DESCRIPTION
**Pull request by jkmassel**
Removes CircleCI – replacing its last use (release builds) with API calls to Buildkite.

### CODE REVIEW

**Comment by jkmassel** [comment]
You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/'>here</a> with build number: <code>pr17891-f1728dd</code>. IPA is available <a href='https://appcenter-filemanagement-distrib1ede6f06e.azureedge.net/26759510-daeb-419a-b7a5-16f660eccbe9/Jetpack%20Alpha.ipa?sv=2019-02-02&sr=c&sig=jPAE8jfLx3uSMDt1tuO0b%2FQ0aCdQh8SDt7GfdGRZUGk%3D&se=2022-02-10T21%3A51%3A11Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Comment by jkmassel** [comment]
You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17891-f1728dd</code>. IPA is available <a href='https://appcenter-filemanagement-distrib4ede6f06e.azureedge.net/2730ec48-804a-4b0a-96ed-5b8349f719d6/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=32yqe3tsyssoFT4%2BA1B9OgBLY55wpbqRcdinQcu5GZo%3D&se=2022-02-10T21%3A52%3A45Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Comment by jkmassel** [comment]
@mokagio – this should be the last piece of the puzzle – once this lands you should have all the Buildkite tooling as part of the next release!

File: README.md
Comment (mokagio) [inline README.md:3]: We could also remove this wholesale, given GitHub reports the build status for each commit these days, something that didn't happen at the time the badges became trendy.

**Review by mokagio — CHANGES_REQUESTED** [review]
Apologies @jkmassel, I just noticed I missed something when reviewing #17862. 

I could be wrong, but it looks to me like there is no way currently to tell the `release-builds.yml` pipeline whether it should build a beta or a stable build. In practice, I'm pretty sure the only difference is in the GitHub release always being a stable one (i.e. not having the "pre-release" checkbox selected) but I think it's something we ought to address.

I know you're AFK today and this work affect me directly, so I'm going to:

- Request changes to avoid this being merged accidentally
- Open a PR against this branch to address this

---

Update: Also, this doesn't use a version of release toolkit that includes the `buildkite_trigger_build` action, yet.

**Comment by mokagio** [comment]
> Open a PR against this branch to address this

https://github.com/wordpress-mobile/WordPress-iOS/pull/17893

**Review by mokagio — APPROVED** [review]
Now that both the fix to differentiate between beta and stable and the release toolkit update have been implemented, we can ship this!

Looking forward test this in action as soon as there'll be a new 19.2 beta to ship 🚀

**Comment by mokagio** [comment]
Ouch!

Auto-merge kicked in because there are no GitHub checks requirements on the release branch 🤔 

image

**Comment by mokagio** [comment]
It looks like Octokit exposes a way to set required checks while setting the branch protection: https://github.com/octokit/octokit.rb/blob/35d968fddf43f8cd7618e082e17e958be67964f1/spec/octokit/client/repositories_spec.rb#L225-L228

Maybe we should look into updating the release toolkit to use that? 🤔

**Comment by jkmassel** [comment]
> Maybe we should look into updating the release toolkit to use that? 🤔

Agreed!!

**Comment by AliSoftware** [comment]
@mokagio that's why I opened pdnsEh-7d-p2 :D Maybe it would be worth bumping the priority of that one a bit, especially given it probably won't take _that much_ time to implement? (famous last words…)

**Pull request merged**
